tiny_dnn/core/params/fully_params.h

```diff
@@ -26,14 +26,22 @@
 */
 #pragma once
 
+#include "params.h"
+
 namespace tiny_dnn {
 namespace core {
 
-struct fully_params {
+class fully_params : public Params {
+ public:
     cnn_size_t in_size_;
     cnn_size_t out_size_;
     bool has_bias_;
 };
 
+// TODO(nyanp): can we do better here?
```

Comment:
@edgarriba 
It works, but a bit curious design because base and derived class depend on each other. We want ```Params``` to just hold any types, but we don't need polymorphism for them. In such case I prefer to use the ```boost::any``` pattern.

http://stackoverflow.com/questions/4988939/how-do-boostvariant-and-boostany-work